### PR TITLE
docs(@nguniversal/builders): Added Builder Readme Overview

### PR DESCRIPTION
**Pull request by conradkirschner**
A start for documentation about the universal builders. 

Better then WIP I would say, for now. Maybe it gives some overview about how the docs could look like.

### CODE REVIEW

File: modules/builders/README.md [hunk 1]
```diff
@@ -1,3 +1,83 @@
...
+Use this if you don't have any server side Code - as "serverTarget" is not configurable.
+
+### Parameter
+
+| Name          | Type of Value                                | Description                                                                       | Required |
+| ------------- | -------------------------------------------- | --------------------------------------------------------------------------------- | -------- |
+| browserTarget | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Target to build.                                                                  | ✔️       |
+| routesFile    | A Filename - Attention no schema validation! | The path to a file containing routes separated by newlines.                       |          |
+| routes        | array of Unique Strings                      | The routes to render.                                                             |          |
+| guessRoutes   | Bolean (default: true)                       | Whether or not the builder should extract routes and guess which paths to render. |          |
```
Comment (alan-agius4) [inline modules/builders/README.md:78]:
```suggestion
| guessRoutes   | Boolean (default: true)                       | Whether or not the builder should extract routes and guess which paths to render. |          |
```

File: modules/builders/README.md [hunk 2]
```diff
@@ -1,3 +1,83 @@
 # Angular Universal Builders
 
+Angular Universal comes with some custom Builders. All Parameters are also specified in their schema.json file.  
+In any case this Readme.md gets outdated - it's worthy to have a look there!
```
Comment (alan-agius4) [inline modules/builders/README.md:4]:
```suggestion
In any case this Readme.md gets outdated - it's worthy to have a look there!
```

Lets remove this note and make sure to keep the doc updated instead.

File: modules/builders/README.md [hunk 3]
```diff
@@ -1,3 +1,83 @@
...
+
+### Parameter
+
+| Name          | Type of Value                                | Description                                                                       | Required |
+| ------------- | -------------------------------------------- | --------------------------------------------------------------------------------- | -------- |
+| browserTarget | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Target to build.                                                                  | ✔️       |
+| routesFile    | A Filename - Attention no schema validation! | The path to a file containing routes separated by newlines.                       |          |
+| routes        | array of Unique Strings                      | The routes to render.                                                             |          |
+| guessRoutes   | Bolean (default: true)                       | Whether or not the builder should extract routes and guess which paths to render. |          |
+| numProcesses  | Number (min: 1, default: 1)                  | The number of cpus to use. Defaults to all but one.                               |          |
```
Comment (alan-agius4) [inline modules/builders/README.md:79]:
```suggestion
| numProcesses  | number (min: 1, default: 1)                  | The number of cpus to use. Defaults to all but one.                               |          |
```
Not consistence.

File: modules/builders/README.md [hunk 4]
```diff
@@ -1,3 +1,83 @@
...
+
+Builds the browser and server, then renders each route in options.routes and writes them to prerender/<route>/index.html for each output path in the browser result.  
+Use this if you don't have any server side Code - as "serverTarget" is not configurable.
+
+### Parameter
+
+| Name          | Type of Value                                | Description                                                                       | Required |
+| ------------- | -------------------------------------------- | --------------------------------------------------------------------------------- | -------- |
+| browserTarget | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Target to build.                                                                  | ✔️       |
+| routesFile    | A Filename - Attention no schema validation! | The path to a file containing routes separated by newlines.                       |          |
```
Comment (alan-agius4) [inline modules/builders/README.md:76]:
```suggestion
| routesFile    | string | The path to a file containing routes separated by newlines.                       |          |
```

There are paths validation in code.

File: modules/builders/README.md [hunk 5]
```diff
@@ -1,3 +1,83 @@
...
+WIP
+
+## Static Generator
+
+### Summary (WIP)
+
+Builds the browser and server, then renders each route in options.routes and writes them to prerender/<route>/index.html for each output path in the browser result.  
+Use this if you don't have any server side Code - as "serverTarget" is not configurable.
+
+### Parameter
```
Comment (alan-agius4) [inline modules/builders/README.md:71]:
```suggestion
### Option
```

File: modules/builders/README.md [hunk 6]
```diff
@@ -1,3 +1,83 @@
...
+| proxyConfig      | string                      | Proxy configuration file.                                                                                                                              |          |
+| waitBeforeReload | number                      | Waiting time before reload the page (in Milliseconds)                                                                                                  |          |
+
+### Example
+
+WIP
+
+## Static Generator
+
+### Summary (WIP)
```
Comment (alan-agius4) [inline modules/builders/README.md:66]:
```suggestion
### Description
```

File: modules/builders/README.md [hunk 7]
```diff
@@ -1,3 +1,83 @@
...
+| ssl              | boolean (default: false)    | Serve using HTTPS.                                                                                                                                     |          |
+| sslKey           | string                      | SSL key to use for serving HTTPS.                                                                                                                      |          |
+| sslCert          | string                      | SSL certificate to use for serving HTTPS.                                                                                                              |          |
+| proxyConfig      | string                      | Proxy configuration file.                                                                                                                              |          |
+| waitBeforeReload | number                      | Waiting time before reload the page (in Milliseconds)                                                                                                  |          |
+
+### Example
```
Comment (alan-agius4) [inline modules/builders/README.md:60]:
```suggestion

```

File: modules/builders/README.md [hunk 8]
```diff
@@ -1,3 +1,83 @@
...
+
+WIP
```
Comment (alan-agius4) [inline modules/builders/README.md:62]:
```suggestion

```

File: modules/builders/README.md [hunk 9]
```diff
@@ -1,3 +1,83 @@
...
+| ssl              | boolean (default: false)    | Serve using HTTPS.                                                                                                                                     |          |
+| sslKey           | string                      | SSL key to use for serving HTTPS.                                                                                                                      |          |
+| sslCert          | string                      | SSL certificate to use for serving HTTPS.                                                                                                              |          |
+| proxyConfig      | string                      | Proxy configuration file.                                                                                                                              |          |
+| waitBeforeReload | number                      | Waiting time before reload the page (in Milliseconds)                                                                                                  |          |
```
Comment (alan-agius4) [inline modules/builders/README.md:58]:
```suggestion

```

File: modules/builders/README.md [hunk 10]
```diff
@@ -1,3 +1,83 @@
...
+
+### Example
+
+WIP
+
+## SSR Dev Server
+
+### Summary
+
+This builder is the replacement for ng serve, if you develop an Angular Universal App.
```
Comment (alan-agius4) [inline modules/builders/README.md:38]:
```suggestion
This builder is the replacement for ng serve, if you develop an Angular Universal application.
```

File: modules/builders/README.md [hunk 11]
```diff
@@ -1,3 +1,83 @@
...
+
+# Builders
+
+- prerender
+- ssr-dev-server
+- static-generator
+
+## Prerender
+
+### Summary
```
Comment (alan-agius4) [inline modules/builders/README.md:14]:
```suggestion
### Description
```

File: modules/builders/README.md [hunk 12]
```diff
@@ -1,3 +1,83 @@
...
+
+This Builder will prerender the routes that you define in a Route-File or as Parameter.
+Use this if you want to serve fast Pages, where you don't have changing Content on _each request_.
+
+### Parameter
```
Comment (alan-agius4) [inline modules/builders/README.md:19]:
```suggestion
### Options
```

File: modules/builders/README.md [hunk 13]
```diff
@@ -1,3 +1,83 @@
...
+| ------------- | -------------------------------------------- | --------------------------------------------------------------------------------- | -------- |
+| browserTarget | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Target to build.                                                                  | ✔️       |
+| serverTarget  | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Server target to use for prerendering the app.                                    | ✔️       |
+| routesFile    | A Filename - Attention no schema validation! | The path to a file containing routes separated by newlines.                       |          |
+| routes        | Unique Strings                               | The routes to render.                                                             |          |
+| guessRoutes   | Bolean (default: true)                       | Whether or not the builder should extract routes and guess which paths to render. |          |
+| numProcesses  | Number (min: 1, default: 1)                  | The number of cpus to use. Defaults to all but one.                               |          |
+
+### Example
```
Comment (alan-agius4) [inline modules/builders/README.md:30]:
```suggestion

```

File: modules/builders/README.md [hunk 14]
```diff
@@ -1,3 +1,83 @@
...
+
+WIP
```
Comment (alan-agius4) [inline modules/builders/README.md:32]:
```suggestion
=
```

File: modules/builders/README.md [hunk 15]
```diff
@@ -1,3 +1,83 @@
...
+
+- prerender
+- ssr-dev-server
+- static-generator
+
+## Prerender
+
+### Summary
+
+This Builder will prerender the routes that you define in a Route-File or as Parameter.
```
Comment (alan-agius4) [inline modules/builders/README.md:16]:
```suggestion
This builder is used to prerender pages of your application. Prerendering is the process where a dynamic page is processed at build time generating static HTML.

For more information about this see: https://angular.io/guide/prerendering

```

File: modules/builders/README.md [hunk 16]
```diff
@@ -1,3 +1,83 @@
...
+### Example
+
+WIP
+
+## SSR Dev Server
+
+### Summary
+
+This builder is the replacement for ng serve, if you develop an Angular Universal App.
+Use this if you want to develop locally an Angular Universal Application
```
Comment (alan-agius4) [inline modules/builders/README.md:39]:
```suggestion
This builder will launch a development server with live-reload capabilities for Angular Universal applications. 
```

File: modules/builders/README.md [hunk 17]
```diff
@@ -1,3 +1,83 @@
...
+It will hot reload only things that you have changed and trigger a refresh in the browser.
```
Comment (alan-agius4) [inline modules/builders/README.md:40]:
```suggestion

```

File: modules/builders/README.md (hunk 10 above)
Comment (alan-agius4) [inline modules/builders/README.md:38]:
```suggestion

```
This is not quite right, this builder isn't intended to replace `ng serve` but rather offer a way to serve and watch a universal application.

File: modules/builders/README.md [hunk 18]
```diff
@@ -1,3 +1,83 @@
...
+
+| Name          | Type of Value                                | Description                                                                       | Required |
+| ------------- | -------------------------------------------- | --------------------------------------------------------------------------------- | -------- |
+| browserTarget | Pattern: `^[^:\s]+:[^:\s]+(:[^\s]+)?$`       | Target to build.                                                                  | ✔️       |
+| routesFile    | A Filename - Attention no schema validation! | The path to a file containing routes separated by newlines.                       |          |
+| routes        | array of Unique Strings                      | The routes to render.                                                             |          |
+| guessRoutes   | Bolean (default: true)                       | Whether or not the builder should extract routes and guess which paths to render. |          |
+| numProcesses  | Number (min: 1, default: 1)                  | The number of cpus to use. Defaults to all but one.                               |          |
+
+### Example
```
Comment (alan-agius4) [inline modules/builders/README.md:81]:
```suggestion

```

File: modules/builders/README.md [hunk 19]
```diff
@@ -1,3 +1,83 @@
...
+
 WIP
```
Comment (alan-agius4) [inline modules/builders/README.md:83]:
```suggestion

```

File: modules/builders/README.md [hunk 20]
```diff
@@ -1,3 +1,83 @@
...
+
+### Example
+
+WIP
+
+## Static Generator
+
+### Summary (WIP)
+
+Builds the browser and server, then renders each route in options.routes and writes them to prerender/<route>/index.html for each output path in the browser result.
```
Comment (alan-agius4) [inline modules/builders/README.md:68]:
```suggestion
This builder is used to prerender pages of your application using the experimental [Clover](../common/clover/README.md) implementation. Prerendering is the process where a dynamic page is processed at build time generating static HTML.

Attention: 
```

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this. 

Please squash your commits.

**Comment by conradkirschner** [comment]
@alan-agius4  Thanks for fast reply and suggestion, I just added them and squashed the commits :)

File: modules/builders/README.md [hunk 21]
```diff
@@ -1,3 +1,76 @@
...
+
+## Static Generator
+
+### Description
+
+This builder is used to prerender pages of your application using the experimental [Clover](../common/clover/README.md) implementation.
+Prerendering is the process where a dynamic page is processed at build time generating static HTML.
+
+Attention:
+Use this if you don't have any server side Code - as "serverTarget" is not configurable.
```
Comment (conradkirschner) [inline modules/builders/README.md:66]: Is that correct?

**Comment by hiepxanh** [comment]
@alan-agius4  I think this docs is full of information, or we just reject ✌️

**Review by alan-agius4 — APPROVED** [review]
LGTM

**Pull request merged**